org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/AxesMetadataExample.java

```diff
@@ -0,0 +1,173 @@
+package org.eclipse.january.examples.dataset;
+
+import java.util.Arrays;
+
+import org.eclipse.january.dataset.CompoundDataset;
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.DatasetFactory;
+import org.eclipse.january.dataset.IndexIterator;
+import org.eclipse.january.dataset.LongDataset;
+import org.eclipse.january.dataset.Maths;
+import org.eclipse.january.metadata.AxesMetadata;
+import org.eclipse.january.metadata.internal.AxesMetadataImpl;
+import org.eclipse.uomo.units.SI;
+
+public class AxesMetadataExample {
+
+	public static void main(String[] args) {
+		testDataset();
+		testCompoundDataset();
+	}
+
+	private static void testDataset() {
+		// configure Dataset A
+		Dataset timestampsA = DatasetFactory.createFromList(Arrays.asList(100l, 200l, 300l));
+		timestampsA.addMetadata(new UomMetadata(SI.SECOND));
+		Dataset speedsA = DatasetFactory.createFromList(Arrays.asList(1d, 2d, 3d));
+		AxesMetadata axesMetadataA = new AxesMetadataImpl();
+		axesMetadataA.initialize(1);
+		axesMetadataA.setAxis(0, timestampsA);
+		speedsA.addMetadata(axesMetadataA);
+		speedsA.addMetadata(new UomMetadata(SI.METRES_PER_SECOND));
+		speedsA.setName("speedsA");
+
+		// configure Dataset B
+		Dataset timestampsB = DatasetFactory.createFromList(Arrays.asList(100l, 200l, 300l));
+		timestampsB.addMetadata(new UomMetadata(SI.SECOND));
+		Dataset speedsB = DatasetFactory.createFromList(Arrays.asList(2d, 4d, 8d));
+		AxesMetadata axesMetadataB = new AxesMetadataImpl();
+		axesMetadataB.initialize(1);
+		axesMetadataB.setAxis(0, timestampsB);
+		speedsB.addMetadata(axesMetadataB);
+		speedsB.addMetadata(new UomMetadata(SI.METRES_PER_SECOND));
+		speedsB.setName("speedsB");
+
+		// show initial values
+		printTimedDataset(speedsA);
+		printTimedDataset(speedsB);
+
+		// non-destructive addition (loses units)
+		Dataset sumAB2 = Maths.add(speedsA, speedsB);
+		Dataset productAB2 = Maths.multiply(speedsA, speedsB);
+
+		// destructive addition (amends speedsA)
+		Dataset sumAB = speedsA.iadd(speedsB);
```

Comment:
@dinkoivanov - note the above iadd() operation is destructive.  The values in B get added to those in A, and then A is returned.  The destructive operation isn't compliant with our "flow of data" model. We need A and B to remain unchanged, with C created to store their result.

`Maths.add(a,b)` is the non destructive equivalent.